misc/services/mobile-webui/mobile-webui-frontend-testing/tests/spec/picking/picking.spec.js

```diff
@@ -118,3 +120,29 @@ test('Scan invalid picking slot QR code', async ({ page }) => {
         await PickingJobScreen.scanPickingSlot({ qrCode: 'invalid QR code' });
     });
 });
+
+// noinspection JSUnusedLocalSymbols
+test('Test picking line complete status - draft | in progress | complete', async ({ page }) => {
+    const { login, pickingSlotQRCode, documentNo, huQRCode, luPIName } = await createMasterdata();
+
+    await LoginScreen.login(login);
+    await ApplicationsListScreen.expectVisible();
+    await ApplicationsListScreen.startApplication('picking');
+    await PickingJobsListScreen.waitForScreen();
+    await PickingJobsListScreen.filterByDocumentNo(documentNo);
+    await PickingJobsListScreen.startJob({ documentNo });
+    await PickingJobScreen.scanPickingSlot({ qrCode: pickingSlotQRCode });
+    await PickingJobScreen.setTargetLU({ lu: luPIName });
+
+    await PickingJobScreen.expectLineStatusColor({ index: 1, color: 'red' });
+
+    await PickingJobScreen.pickHU({ qrCode: huQRCode, qtyEntered: '2', expectQtyEntered: '3', qtyNotFoundReason: QTY_NOT_FOUND_REASON_IGNORE });
+
+    await PickingJobScreen.expectLineStatusColor({ index: 1, color: 'yellow' });
+
+    await PickingJobScreen.pickHU({ qrCode: huQRCode, expectQtyEntered: '1' });
+
+    await PickingJobScreen.expectLineStatusColor({ index: 1, color: 'green' });
+
+    await PickingJobScreen.complete();
```

Comment:
how exactly are we testing this feature?
imho
* createMasterdata shall take { allowCompletingPartialPickingJob } = {} as param
* have here 2 tests grouped under one `test.describe`, one is test is checking the behaviour of allowCompletingPartialPickingJob=false when partial picking. The other test, using same scenario is testing the behaviour of allowCompletingPartialPickingJob=true when partial picking.

Reply:
yep, I agree 👍